GAE/src/org/akvo/flow/xml/XmlQuestionGroup.java

```diff
@@ -0,0 +1,108 @@
+/*
+ *  Copyright (C) 2019 Stichting Akvo (Akvo Foundation)
+ *
+ *  This file is part of Akvo FLOW.
+ *
+ *  Akvo FLOW is free software: you can redistribute it and modify it under the terms of
+ *  the GNU Affero General Public License (AGPL) as published by the Free Software Foundation,
+ *  either version 3 of the License or any later version.
+ *
+ *  Akvo FLOW is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ *  without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *  See the GNU Affero General Public License included below for more details.
+ *
+ *  The full license text can also be seen at <http://www.gnu.org/licenses/agpl.html>.
+ */
+
+package org.akvo.flow.xml;
+
+import java.util.Arrays;
+import java.util.TreeMap;
+
+import org.waterforpeople.mapping.app.gwt.client.survey.QuestionDto;
+import org.waterforpeople.mapping.app.gwt.client.survey.QuestionGroupDto;
+import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlElementWrapper;
+import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlProperty;
+
+/* Class for working with XML like this:
+<questionGroup repeatable = "false">
+    <heading>G2</heading>
+    <question>
+    </question>
+</questionGroup>
+ */
+public class XmlQuestionGroup {
+
+    @JacksonXmlElementWrapper(useWrapping = false)
+    private XmlQuestion[] question;
+
+    @JacksonXmlProperty(localName = "heading", isAttribute = false)
+    private String heading;
+
+    @JacksonXmlProperty(localName = "repeatable", isAttribute = true)
+    private boolean repeatable;
+
+    private int order; //Not in XML
```

Comment:
If its not there then do we need it here?

Reply:
Yes. It is filled in from the XmlQuestion class and then used when making the Dto.

Reply:
Alternatively I suppose it could be a parameter of XmlQuestionGroup.toDto()